server/src/main/java/org/opensearch/index/mapper/KnnContext.java

```diff
@@ -0,0 +1,117 @@
+/*
+ *  Copyright OpenSearch Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.index.mapper;
+
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.common.xcontent.ToXContentFragment;
+import org.opensearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Abstracts KNN segment of dense_vector field type
+ */
+public class KnnContext implements ToXContentFragment, Writeable {
```

Comment:
Can we mark this class as final to prevent any subclassing fragmentation?

```suggestion
public final class KnnContext implements ToXContentFragment, Writeable {
```

Reply:
ack